src/mlpack/methods/preprocess/preprocess_scale_main.cpp

```diff
@@ -0,0 +1,174 @@
+    "scaler_method", "pcawhitening", "epsilon", 0.01) +
+    "\n\n"
+    "You can also retransform the scaled dataset back using" +
+    PRINT_PARAM_STRING("function") + ". An example to rescale : " +
+    PRINT_DATASET("X_scaled") + " into " + PRINT_DATASET("X")
+    + "using the saved model " + PRINT_PARAM_STRING("input_model") + " is:"
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X_scaled", "output", "X",
+    "function", 1, "input_model", "saved")+
+    "\n\n"
+    "Another simple example where we want to scale the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_scaled") + " with "
+    " min_max_scaler as scaler method, where scaling range is 1 to 3 instead"
+    " of deafult 0 to 1. We coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "min_max_scaler", "min_value", 1, "max_value", 3),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// Define parameters for data.
+PARAM_MATRIX_IN_REQ("input", "Matrix containing data.", "i");
+PARAM_MATRIX_OUT("output", "Matrix to save scaled data to.", "o");
+PARAM_STRING_IN("scaler_method", "method to use for scaling, the "
+    "default is standard_scaler.", "a", "standard_scaler");
+PARAM_DOUBLE_IN("epsilon", "regularization Parameter for pcawhitening,"
+    " or zcawhitening, should be between -1 to 1.", "r", 0.000001);
+
+PARAM_INT_IN("seed", "Random seed (0 for std::time(NULL)).", "s", 0);
+PARAM_INT_IN("min_value", "Starting value of range for min_max_scaler.",
+    "b", 0);
+PARAM_INT_IN("max_value", "Ending value of range for min_max_scaler.",
+    "e", 1);
+PARAM_INT_IN("function", "function to apply,either 0:Transform or 1:Inverse"
+  "Transform", "f", 0)
+// Loading/saving of a model.
+PARAM_MODEL_IN(ScalingModel, "input_model", "Input Scaling model.", "m");
+PARAM_MODEL_OUT(ScalingModel, "output_model", "Output scaling model.", "M");
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  const std::string scalerMethod = CLI::GetParam<string>("scaler_method");
+
+  if (CLI::GetParam<int>("seed") == 0)
+    mlpack::math::RandomSeed(std::time(NULL));
+  else
+    mlpack::math::RandomSeed((size_t) CLI::GetParam<int>("seed"));
+  // Make sure the user specified output filenames.
+  RequireAtLeastOnePassed({ "output", "output_model"}, false,
+      "no output will be saved");
+  // Check scaler method.
+  RequireParamValue<std::string>("scaler_method",
+      [](std::string x) { return x == "standard_scaler" || x ==
+      "min_max_scaler" || x == "mean_normalization" || x == "max_abs_scaler" ||
+      x == "whitening";}, true, "scaler_method must be one among min_max_"
+      "scaler, max_abs_scaler, pcawhitening, standard_scaler, zcawhitening"
+      " or mean_normalization.");
```

Comment:
You could use `RequireParamInSet()` here.

Reply:
i didn't knew about it, Thanks for letting me know.